lib/db.js

```diff
@@ -37,6 +37,7 @@ module.exports = (
 
   const features = require('./features')(config)
   const redis = require('./redis')(config.redis, log)
+  const safeUrl = require('./safe-url')(log)
```

Comment:
Would moving this validation inside to `/lib/pool.js` make more sense? Possibly passing in params

```javascript
Pool.prototype.post = function (path, params, data) {..}
```

Reply:
Hmm, maybe. I did consider calling it inside `lib/pool`, but decided against it for the following reasons:

1. To make the logging more useful if these ever occur in prod, I pass in the name of the calling method so we can immediately pinpoint the source of the bad param. Adding that to the `pool` methods as an extra arg felt like pushing it to the wrong place, and requires updating unrelated code.

2. Not every call to a `pool` method needs to use `safeUrl`, only the ones where we're interpolating values. That seems like a decision that belongs more in the `db` module to me, although maybe I'm not looking at it right.

3. If I move it into `pool` then the local tests for `db` and `pool` both need to change. If I keep it in `db` they can stay the same. So less typing, less reviewing, less chance of me breaking something when I edit the tests. 😄

But I can see that one reason in favour of doing it in `pool` is so that callers can't forget to use `safeUrl`. We could also guard against that with appropriate assertions in `test/local/db.js`, which I should probably have added before I opened this PR but I didn't think of until now.

Would extra local test assertions relieve your concern, or is it based on something else? Does any of my reasoning above make sense or am I focusing on the wrong things?

Reply:
> To make the logging more useful if these ever occur in prod, I pass in the name of the calling method so we can immediately pinpoint the source of the bad param.

But of course, I wouldn't need to pass the name in from `pool` because it knows the HTTP method, which is enough to trace back to the caller from (in combination with the path, obviously). Okay, I am persuaded, closing this while I refactor.

Reply:
> ...but decided against it for the following reasons...

Really, I could have replace all of that verbiage with "because I'm lazy".

Reply:
Actually I've changed my mind about this. Moving it in to `pool` doesn't do anything to stop callers from forgetting to use `safeUrl` because they can still just pass in a vanilla path without params to interpolate. All it does is make the change bigger and hairier, without adding value.

Re-opening this in its original state, although I'm still open-minded about it if there's a good reason to prefer it the other way.

Reply:
> doesn't do anything to stop callers from forgetting to use safeUrl

I think it just felt right to me to have the url validation live as close as possible to issuing the request and have it automagically do the right thing. I can definitely see myself forgetting to call `safeUrl`. 

If the refactor is really hairy, then it might not be worth the change. We could group this and encoding of email params into a more generic refactor bug.

Reply:
> I can definitely see myself forgetting to call `safeUrl`.

It might be that I just don't properly understand your suggestion yet. What is to stop you forgetting if the call to `safeUrl` lives inside `pool`? What does e.g. the implementation for `pool.post` look like and what arguments is the caller passing in?

As it stands, I think both approaches require explicit tests that contain unsafe characters. I'll have a crack at those tomorrow, although making sure every db method is covered will also, I suspect, involve writing tests for some methods that don't have proper coverage yet (which is useful in it's own right of course). But I don't see a way round it with either approach.